modules/nf-core/picard/collecthsmetrics/main.nf

```diff
@@ -26,13 +27,33 @@ process PICARD_COLLECTHSMETRICS {
     def prefix = task.ext.prefix ?: "${meta.id}"
     def reference = fasta ? "--REFERENCE_SEQUENCE ${fasta}" : ""
 
+    def args2 = task.ext.args2 ?: ''
+    def reference_dict = "--SEQUENCE_DICTIONARY ${dict}"
+    if (bait_intervals.toString() ==~ ".*bed.*" || target_intervals.toString() ==~ ".*bed.*"){
+        if ( ! dict ){
+            exit 1, "Sequence dictionary input is mandatory when using bed intervals."
+        }
+        bait_intervals = bait_intervals.toString() - ~/.bed.*$/ + ".interval_list"
+        target_intervals = target_intervals.toString() - ~/.bed.*$/ + ".interval_list"
+    }
+
     def avail_mem = 3
     if (!task.memory) {
         log.info '[Picard CollectHsMetrics] Available memory not known - defaulting to 3GB. Specify process memory requirements to change this.'
     } else {
         avail_mem = task.memory.giga
     }
     """
+    for f in \$(find -name "*.bed*"); do
+    picard \\
+        -Xmx${avail_mem}g \\
+        BedToIntervalList \\
+        $args2 \\
+        --INPUT \$f \\
+        --OUTPUT \${f%.bed*}.interval_list \\
+        ${reference_dict}
```

Comment:
I would indent this to make it clearer what is being used within the for loop.

Also does anything happen if no bed files are supplied?

Reply:
Also, this looks rather 'custom', shouldn't this be from a separate module, if the CollectHsMetrics doesn't accept the `bed` format directly? 

Modules shouldn't peform extra 'functionality' for the user

Reply:
I've added the indentation. If no bed files are supplied the for loop is ignored. 

Yeah, the main reason I wanted a through review is because of the "added" functionality. IMO, I'm just adding another input option, but I can see how that would go against the guidelines.

Reply:
Yeah, then I would lean to saying 'no' here... but would be good to get the opinions from other @nf-core/modules-team